agents/appmotion_motor/appmotion_motors_agent.py

```diff
@@ -0,0 +1,925 @@
+            self.motors.stop_jogging(motor)
+
+        return True, "Stopped jogging motor {}".format(motor)
+
+    def seek_home_linear_stage(self, session, params=None):
+        """seek_home_linear_stage(motor=1):
+        
+        **Task** - Move the linear stage to its home position (using the home
+        limit switch).
+            
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+        """
+
+        @ocs_agent.param('motor', default=1, type=int)
+        
+        self.move_status = self.is_moving(motor)
+        if self.move_status:
+            return False, "Motors are already moving."
+
+        with self.lock.acquire_timeout(timeout=1, job=f'seek_home_linear_stage_motor{motor}') as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not seek_home_linear_stage because lock held by {self.lock.job}")
+                return False, "Could not acquire lock"
+            self.motors.seek_home_linear_stage(motor)
+
+        return True, "Moving motor {} to home".format(motor)
+
+    def set_zero(self, session, params=None):
+        """set_zero(motor=1):
+        
+        **Task** - Sets the zero position (AKA home) for motor(s) specified in
+        params.
+        
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+        """
+
+        @ocs_agent.param('motor', default=1, type=int)
+        
+        self.move_status = self.is_moving(motor)
+        if self.move_status:
+            return False, "Motors are already moving."
+
+        with self.lock.acquire_timeout(1, job=f"set_zero_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f'Could not set_zero because lock held by {self.lock.job}')
+                return False, 'Could not acquire lock'
+            self.motors.set_zero(motor)
+
+        return True, "Zeroing motor {} position".format(motor)
+
+    def run_positions(self, session, params=None):
+        """run_positions(pos_data=[1,1], motor=1, pos_is_inches=False):
+        
+        **Task** - Runs a tab-delimited list of entries as positions. For 
+        motor=ALL, the first column must be the x-data, and the second column
+        the y-data. Each position will be attained.
+            
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+            pos_data (list): Tab-delimited list of entries. First column
+                is x-data, second column is y-data.
+            pos_is_inches (bool): True if pos was specified in inches, False if
+                in counts (default False)
+        """
+
+        @ocs_agent.param('motor', default=1, type=int)
+        @ocs_agent.param('pos_is_inches', default=True, type=bool)
```

Comment:
A few things here:

1. This still discusses "tab-delimited" lists. That description should be updated.
2. The method signature override should contain the default values for the parameters or `None` if it's a required paramter. `pos_data` is required here, so the method signature override should read: `run_positions(pos_data=None, motor=1, pos_is_inches=False)`.
3. pos_data is missing a param decorator. If you're using the decorators you must describe all possible parameters. OCS will block undefined parameters when trying to pas them in a client. So it should look something like:

```
    @ocs_agent.param('pos_data', type=list)
    @ocs_agent.param('motor', default=1, type=int)
    @ocs_agent.param('pos_is_inches', default=True, type=bool)
    def run_positions(self, session, params=None):
        """run_positions(pos_data=None, motor=1, pos_is_inches=False):
        
        **Task** - Runs a given list of entries as positions. For 
        motor=ALL, the first column must be the x-data, and the second column
        the y-data. Each position will be attained.
            
        Parameters:
            pos_data (list): Specified position [x, y].
            motor (int, optional): Determines which motor, either 1 or 2, 3 is for all
                motors. (default 1)
            pos_is_inches (bool, optional): True if pos was specified in inches, False if
                in counts (default False)
        """
```